File: sdk/agid-kotlin/src/main/kotlin/org/agid/Agid.kt
package org.agid

const val BASE32_ALPHABET = "0123456789ABCDEFGHJKMNPQRSTVWXYZ"
const val AGID_PREFIX_LENGTH = 2
const val AGID_HASH_LENGTH = 10
const val AGID_TOTAL_LENGTH = 12

data class AgidResult(
    val id: String,
    val lat: Double,
    val lon: Double,
    val face: Int? = null,
)

data class AgidBounds(
    val minLat: Double,
    val maxLat: Double,
    val minLon: Double,
    val maxLon: Double,
)

fun encode(lat: Double, lon: Double): AgidResult {
    throw NotImplementedError("wire this package to the AGID reference implementation")
}

fun decode(id: String): AgidResult? = null

fun cellBounds(id: String): AgidBounds {
    throw NotImplementedError("wire this package to the AGID reference implementation")
}

File: sdk/agid-dart/lib/agid.dart
const base32Alphabet = '0123456789ABCDEFGHJKMNPQRSTVWXYZ';
const agidPrefixLength = 2;
const agidHashLength = 10;
const agidTotalLength = 12;

class AgidResult {
  const AgidResult({required this.id, required this.lat, required this.lon, this.face});

  final String id;
  final double lat;
  final double lon;
  final int? face;
}

class AgidBounds {
  const AgidBounds({required this.minLat, required this.maxLat, required this.minLon, required this.maxLon});

  final double minLat;
  final double maxLat;
  final double minLon;
  final double maxLon;
}

AgidResult encode(double lat, double lon) {
  throw UnimplementedError('wire this package to the AGID reference implementation');
}

AgidResult? decode(String id) => null;

AgidBounds cellBounds(String id) {
  throw UnimplementedError('wire this package to the AGID reference implementation');
}
